torch/_inductor/fx_passes/joint_graph.py

```diff
@@ -432,12 +433,94 @@ def constant_fold_uniform_value(gm: torch.fx.GraphModule):
         remove_redundant_views(gm)
 
 
+def canonicalize_quant_mapping(gm: torch.fx.GraphModule):
+    """
+    To reuse code and simplify tracing, we trace invoke_quant as invoke_subgraph through the
+    joint graph. Then, we map invoke_quant which we traced as invoke_subgraph back to
+    torch.higher_order_ops.invoke_quant. We removed the "identifier" arg, add back the
+    scheme for ease of pattern matching, add the quant options as a meta on the invoke_quant node,
+    and unwrap single element tuples.
+
+    torch.ops.higher_order.invoke_subgraph(repeated_subgraph0, 'quant_invoke_0_0', (arg0_1, arg1_1));
+    ->
+    torch.ops.higher_order.invoke_quant(repeated_subgraph0, arg0_1, arg1_1, scheme = 'nf4');
+    """
+    graph = gm.graph
+    subgraph_invocations = graph.find_nodes(
+        op="call_function", target=torch.ops.higher_order.invoke_subgraph
+    )
+    tc = torch._guards.TracingContext.try_get()
+    if not tc:
+        return
+    invoke_subgraph_mapping = tc.invoke_subgraph_mapping
```

Comment:
This seems a bit janky 🤔

Reply:
maybe - but would prefer we discuss that on the dynamo pr below below